wandb/cli/cli.py

```diff
@@ -996,41 +1014,59 @@ def launch_sweep(
             wandb.termerror(f"Failed to load job. Error: {e}")
             return False
 
+    scheduler_job = scheduler_args.get("job")
+    if scheduler_job:
+        wandb.termwarn(
+            "Using a scheduler job for launch sweeps is *experimental* and may change without warning"
+        )
+        # validate scheduler job existence
+        try:
+            public_api = PublicApi()
+            public_api.artifact(scheduler_args["job"], type="job")
+        except Exception as e:
+            wandb.termerror(f"Failed to load scheduler job. Error: {e}")
+            return False
+
+    entrypoint = Scheduler.ENTRYPOINT if not scheduler_job else None
     args = sweep_utils.construct_scheduler_args(
+        sweep_type=_type,
+        is_job="job" in scheduler_args,
         sweep_config=parsed_sweep_config,
         queue=queue,
         project=project,
         author=entity,
     )
     if not args:
-        # error already logged
         return
 
-    overrides = {
-        "overrides": {
-            "run_config": {
-                "scheduler": scheduler_args,
-                "launch": launch_args,
-            },
-            "args": args,
-        }
-    }
+    overrides = {"run_config": {}}
+    if launch_args:
+        overrides["run_config"]["launch"] = launch_args
+    if scheduler_args:
+        overrides["run_config"]["scheduler"] = scheduler_args
+    if custom_config:
+        overrides["run_config"]["custom"] = custom_config
```

Comment:
Is custom the best name for this key? What exactly is going in there?

Reply:
i hate custom, its just the word that was built into the sweeps engine for `method`. This config contains any parameters that have to be passed to the scheduler, that aren't super-params like `num_workers` or `polling_sleep`. For example, when using the optuna scheduler, the custom config contains all of the optuna related configuration. Previously, I had it under an `optuna` key, but as you pointed out then we get into that gross top-level-key parsing thing. Any thoughts for alternative designs? @KyleGoyette